### PR TITLE
net-vpn/i2p: add 2.2.1

### PR DESCRIPTION
**Pull request by tharvik**
* quite straight forward version bump
* patch junit tests
  * do haltonerror
  * also doesn't take into account the classpath attribute, only nested classpath; maybe one of you have better way to add a nested tag than maintaining a patch?
* absorbe net-libs/nativebiginteger

also for 2.1.0
 * fix building w/ icedtea; no revbump as unbuildable in this particular mix of JDK

### CODE REVIEW

File: net-vpn/i2p/files/2.2.1-fix-junit-classpath.patch [hunk 1]
```diff
@@ -0,0 +1,122 @@
+diff --git a/apps/i2ptunnel/java/build.xml b/apps/i2ptunnel/java/build.xml
+index b8d8b24a9..cfcf2d5a1 100644
```
Comment (vaukai) [inline net-vpn/i2p/files/2.2.1-fix-junit-classpath.patch:2]: Remove all lines starting with "diff" or "index" (see https://devmanual.gentoo.org/ebuild-writing/misc-files/patches/#clean-patch-howto)

File: net-vpn/i2p/i2p-2.2.1.ebuild [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+	dev-java/json-smart:1
+	dev-java/jsonrpc2-base:1
+	dev-java/jsonrpc2-server:1
+	dev-java/jstl:0
+	dev-java/jstl-api:0
+	dev-java/minidns-core:1
+	dev-java/zxing-core:3
+	dev-java/zxing-javase:3
+	sys-devel/gettext:0[java]
+	>=www-servers/tomcat-9.0.73:9
```
Comment (vaukai) [inline net-vpn/i2p/i2p-2.2.1.ebuild:42]: www-servers/tomcat:9

File: net-vpn/i2p/i2p-2.2.1.ebuild [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+		-e 's,net\.i2p\.crypto\.elgamal\.spec,org.bouncycastle.jce.spec,' \
+		-i core/java/src/net/i2p/crypto/{provider/I2PProvider,CryptoConstants}.java ||
+		die 'redirect imports of bcprov'
+	# getopt, gettext
+	rm -r core/java/src/gnu/{getopt,gettext} || die 'unbundle GNU code'
+	# httpcomponents-client
+	rm -r core/java/src/net/i2p/apache || die 'unbundle httpcomponents-client'
+	sed -e 's,net\.i2p\.apache,org.apache,' \
+		-i core/java/src/net/i2p/util/{Addresses,I2PSSLSocketFactory}.java \
+		   apps/i2pcontrol/java/net/i2p/i2pcontrol/HostCheckHandler.java ||
```
Comment (vaukai) [inline net-vpn/i2p/i2p-2.2.1.ebuild:140]: Replace "   "  with tab

File: net-vpn/i2p/i2p-2.2.1.ebuild [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+src_test() {
+	# avoid rebuilding
+	sed -e '/<delete dir=".\/build" \/>/d' -i core/java/build.xml ||
+		die 'avoid building twice'
+
+	# halt on error
+	find -name build.xml \
+		 -execdir sed -e 's/<junit /\0haltonerror="yes" /' -i {} + ||
+		 die 'ensure test failures propagate'
+
```
Comment (vaukai) [inline net-vpn/i2p/i2p-2.2.1.ebuild:224]: Replace "		 " with "		" in lines 222-223.

**Comment by thesamesam** [comment]
I'm going to merge this now and let @tharvik address the comments in a follow-up or on the next bump, given this fixes some network-related issues.

**Pull request closed without merging**